File: 908-middle-of-the-linked-list/middle-of-the-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        ListNode ft=head,st=head;
        while(ft!=null ||ft.next==null){
            ft=ft.next;
            if(ft!=null){
            ft=ft.next;
            st=st.next;
            }
            if(ft==null ||ft.next==null)
              return st;
        }
        return st;
    }
}

File: 908-middle-of-the-linked-list/middle-of-the-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def middleNode(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        slow = fast = head

        while(fast != None and fast.next != None):
            slow = slow.next
            fast = fast.next.next
        
        return slow
        